File: TH_Buoi1/InsertionSort_Bai1.cpp
#include <iostream>
#include <vector>

void InsertionSort(std::vector<int>& A) {
    for (size_t i = 1; i < A.size(); ++i) {
        int key = A[i];
        int j = static_cast<int>(i) - 1;
        while (j >= 0 && A[j] > key) {
            A[j + 1] = A[j];
            --j;
        }
        A[j + 1] = key;
    }
}

int main() {
    std::vector<int> A = {5, 2, 9, 1, 5, 6};
    InsertionSort(A);
    std::cout << "Mang sau khi sap xep: ";
    for (const auto& num : A) {
        std::cout << num << " ";
    }
    std::cout << std::endl;
    return 0;
}

File: TH_Buoi1/InsertionSort_Bai1.jl
function InsertionSort!(A::Vector{Int})
    for i in 2:length(A)
        key = A[i]
        j = i - 1
        while j >= 1 && A[j] > key
            A[j + 1] = A[j]
            j -= 1
        end
        A[j + 1] = key
    end
    return A
end

function main()
    A = [5, 2, 9, 1, 5, 6]
    InsertionSort!(A)
    println("Mảng sau khi sắp xếp: ", A)
end

main()